lib/openhab/dsl/types/datetime.rb

```diff
@@ -0,0 +1,339 @@
+        #
+        def to_f
+          zoned_date_time.to_epoch_second + zoned_date_time.nano / 1_000_000_000
+        end
+
+        #
+        # The ZonedDateTime representing the state
+        #
+        # @return [Java::java::time::ZonedDateTime] ZonedDateTime representing the state
+        #
+        def zoned_date_time
+          @datetime.zonedDateTime
+        end
+
+        alias to_zdt zoned_date_time
+
+        #
+        # The offset in seconds from UTC
+        #
+        # @return [Integer] The offset from UTC, in seconds
+        #
+        def utc_offset
+          zoned_date_time.offset.total_seconds
+        end
+
+        #
+        # Returns true if time represents a time in UTC (GMT)
+        #
+        # @return [Boolean] true if utc_offset == 0, false otherwise
+        #
+        def utc?
+          utc_offset.zero?
+        end
+
+        #
+        # The timezone
+        #
+        # @return [String] The timezone in `[+-]hh:mm(:ss)` format ('Z' for UTC) or nil if the Item has no state
+        #
+        def zone
+          zoned_date_time.zone.id
+        end
+
+        #
+        # Check if missing method can be delegated to other contained objects
+        #
+        # @param [String, Symbol] meth the method name to check for
+        #
+        # @return [Boolean] true if DateTimeType, ZonedDateTime or Time responds to the method, false otherwise
+        #
+        def respond_to_missing?(meth, *)
+          @datetime.respond_to?(meth) ||
+            zoned_date_time.respond_to?(meth) ||
+            Time.instance_methods.include?(meth.to_sym)
+        end
+
+        #
+        # Forward missing methods to the OpenHAB DateTimeType, its ZonedDateTime object or a ruby Time
+        # object representing the same instant
+        #
+        # @param [String] meth method name
+        # @param [Array] args arguments for method
+        # @param [Proc] block <description>
+        #
+        # @return [Object] Value from delegated method in OpenHAB NumberItem
+        #
+        def method_missing(meth, *args, &block)
+          if @datetime.respond_to?(meth)
+            @datetime.__send__(meth, *args, &block)
+          elsif zoned_date_time.respond_to?(meth)
+            zoned_date_time.__send__(meth, *args, &block)
+          elsif Time.instance_methods.include?(meth.to_sym)
+            to_time.send(meth, *args, &block)
+          else
+            raise NoMethodError, "undefined method `#{meth}' for #{self.class}"
+          end
+        end
+
+        private
+
+        #
+        # Converts other objects to a DateTimeType
+        #
+        # @param [String, Numeric, Time] datetime an object that can be parsed or converted into
+        #   a DateTimeType
+        #
+        # @return [Java::org::openhab::core::library::types::DateTimeType] Object representing the same time
+        #
+        def datetime_type_from(datetime)
```

Comment:
Should these be static methods?

Reply:
As above, these methods are used by the initialize method to construct a DateTimeType from different objects passed to initialize. But I might be able to make them static and call them directly from the <=>, + and - methods which are the only ones that use this I believe.